### PR TITLE
UV Pip Compile increase test coverage

### PR DESCRIPTION
**Pull request by albertferras-vrf**
**Changes:**
- New `uv_pip_compile_file_updater_spec.rb` to test everything that was tested with pip-tool's `pip-compile` command. Reviewed tests one by one and either deleted them or adapted it. (See notes below)
- Bump `uv` to latest version. Tests need minimum 0.2.34 because it adds support for `--no-binary`
- New `uv_pip_compile_options_from_compiled_file` because some options' default are different (--skip-extras, --allow-safe and --emit-build-options). Separate methods make it easier to maintain uv and pipcompile versions.

**Notes:**
**Tests removed:**
- Incompatible versions. These tests used requirements.txt with `—resolver legacy` which does not apply to `uv`. Without the legacy resolver override there is no error on pip-compile or uv pip compile.
- All tests using `--resolver` options
- Tests using custom headers that include stuff like `make upgrade`. These are not relevant for `uv pip compile` mode
- Editable dependencies (`-e giturl`). As of now, uv does not support VCS Urls in editable dependencies. See https://github.com/astral-sh/uv/issues/5442 

**Some differences/changes:**
- "Via setup": `uv` outputs `-e .` and pip compile `-e [file://somepath/dependabot_xxxx`] (file://somepath/dependabot_xxxx%60) in the temporary file. The updated requirements.txt does not change though
- “Unsafe dependencies” (pip and setuptools packages) are removed by pip-compile by default, but on the next major release they will no longer remove those by default (—allow-unsafe as new default). Changed the test to check that uv actually includes `setuptools`, unlike the current pip-compile. See more at  https://github.com/astral-sh/uv/issues/1353

### CODE REVIEW

**Comment by albertferras-vrf** [comment]
hey @avilaton do you think we can merge this on your dependabot's PR? :)

File: python/lib/dependabot/python/file_updater/pip_compile_file_updater.rb [hunk 1]
```diff
@@ -490,6 +490,30 @@ def pip_compile_options_from_compiled_file(requirements_file)
...
+
+          options << "--pre" if requirements_file.content.include?("--pre")
+
+          options << "--no-strip-extras" if requirements_file.content.include?("--no-strip-extras")
+
+          options << "--emit-build-options" if (requirements_file.content.include?("--no-binary") || requirements_file.content.include?("--only-binary"))
+
+          if (resolver = RESOLVER_REGEX.match(requirements_file.content))
+            options << "--resolver=#{resolver}"
+          end
```
Comment (avilaton) [inline python/lib/dependabot/python/file_updater/pip_compile_file_updater.rb:512]: We are going to get a complain about the class length with this

**Review by avilaton — APPROVED** [review]

**Comment by avilaton** [comment]
> hey @avilaton do you think we can merge this on your dependabot's PR? :)

We should add your comments to the other pr in dependabot

**Comment by albertferras-vrf** [comment]
thank you! Yes I already added the comments 👍

**Pull request merged**